bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/LongPolling.java

```diff
@@ -239,6 +239,10 @@ private void handleLongPollResponse(BoschHttpClient httpClient, String subscript
             this.handleFailure.accept(
                     new LongPollingFailedException("Could not deserialize long poll response: '" + content + "'", e));
             return;
+        } catch (Exception e) {
```

Comment:
If external code (out of our control) is involved here, it makes sense to catch everything. I didn't go through the whole flow, but do you know where the `NullPointerException` occurs, and if it's possible to avoid it in the first place?

Reply:
The reported `NullPointerException` is fixed in this PR, among potential other NPEs. I chose to catch all exceptions because of the following reasons:
* We cannot guarantee that other exceptions (including NPEs) occur while processing the results due to bugs in our code
* External code is also involved (Google GSON) that might throw other exceptions as the already caught `JsonSyntaxException` (NPEs etc. are less probable)
* If we don't catch exceptions here, the long polling will die completely. With the current implementation, a new long polling attempt will be initiated after a delay of 15 seconds. Worst case could be a loop of the same systematic long poll error happening every ~15 seconds (but this was also the case previously, at least in case of `JsonSyntaxExceptions`. What do you think?

Reply:
I guess I was confused by the test `testHandleLongPollResponseNPE`, because it seems to simulate a `NullPointerException` being caught here. But this is actually not really expected anymore? I will have another look to understand the original cause of the NPE.

Reply:
Okay, I think I got it: The NPE was thrown from `BoschServiceDataDeserializer`, but this is now fixed. So the mentioned testcase is kind of fictive.

Reply:
Correct 🙂 👍